caddyhttp/requestid/requestid.go

```diff
@@ -0,0 +1,35 @@
+package requestid
+
+import (
+	"context"
+	"log"
+	"net/http"
+
+	"github.com/mholt/caddy"
+	"github.com/mholt/caddy/caddyhttp/httpserver"
+	uuid "github.com/nu7hatch/gouuid"
+)
+
+// Handler is a middleware handler
+type Handler struct {
+	Next httpserver.Handler
+}
+
+func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error) {
+	reqid := UUID()
+	c := context.WithValue(r.Context(), caddy.CtxKey("request_id"), reqid)
```

Comment:
I think @mholt was hoping that there might be a constant created like there is here

https://github.com/mholt/caddy/blob/d5371aff2265540fb8600541bfdf3c3c4fc963e4/caddyhttp/httpserver/middleware.go#L207

    RequestID caddy.CtxKey = "requestid"

then you can do this

    c := context.WithValue(r.Context(), RequestID, reqid)